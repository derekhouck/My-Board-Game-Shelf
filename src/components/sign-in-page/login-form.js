import React from 'react';
import { Field, reduxForm, focus } from 'redux-form';
import { connect } from 'react-redux';
import { login } from '../../actions/auth';
import { required, nonEmpty } from '../../validators';

import Input from '../atoms/input';
import Button from '../button';

export class LoginForm extends React.Component {
    onSubmit(values) {
        return this.props.dispatch(login(values.username, values.password));
    }

    render() {
        const isDisabled = () => this.props.pristine || this.props.submitting;
        let error, body;
        if (this.props.error) {
            error = (
                <div className="form-error" aria-live="polite">
                    {this.props.error.message}
                </div>
            );
        }
        if (this.props.loading) {
            body = (
                <div className="message message-default">
                    Logging you in...
                </div>
            );
        } else {
            body = (
                <form
                    className="login-form"
                    onSubmit={this.props.handleSubmit(values =>
                        this.onSubmit(values)
                    )}>
                    {error}
                    <fieldset>
                        <legend>Account details</legend>
                        <Field
                            component={Input}
                            type="text"
                            name="username"
                            id="username"
                            label="Username"
                            validate={[required, nonEmpty]}
                        />
                        <Field
                            component={Input}
                            type="password"
                            name="password"
                            id="password"
                            label="Password"
                            validate={[required, nonEmpty]}
                        />
                    </fieldset>
                    <Button
                        label="Sign in"
                        disabled={isDisabled()}
                        primary
                    />
                </form>
            );
        }
        return body;
    }
}

const mapStateToProps = state => ({
    error: state.auth.error,
    loading: state.auth.loading
});

export default reduxForm({
    form: 'login',
    onSubmitFail: (errors, dispatch) => dispatch(focus('login', 'username'))
})(connect(mapStateToProps)(LoginForm));