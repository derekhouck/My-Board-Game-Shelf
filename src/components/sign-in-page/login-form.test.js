import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm } from './login-form';

describe('<LoginForm />', function () {
  const handleSubmit = jest.fn();
  const error = { message: 'Test message' };

  it('renders without crashing', () => {
    shallow(<LoginForm handleSubmit={handleSubmit} />);
  });

  it('shows an error when error exists', function () {
    const wrapper = shallow(<LoginForm handleSubmit={handleSubmit} error={error} />);
    expect(wrapper.exists('.form-error')).toBe(true);
  })

  it('shows a message when loading', function () {
    const wrapper = shallow(<LoginForm handleSubmit={handleSubmit} loading={true} />);
    expect(wrapper.exists('.message-default')).toBe(true);
  });
});
