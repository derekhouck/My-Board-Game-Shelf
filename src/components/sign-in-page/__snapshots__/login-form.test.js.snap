// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders without crashing 1`] = `ShallowWrapper {}`;
